bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/auth/client/oauth2/OAuthFactory.java

```diff
@@ -28,6 +29,7 @@
      * Creates a new oauth service. Use this method only once to obtain a handle and store
```

Comment:
the JavaDoc does not match the methods signature anymore.

Reply:
Possible the method name isn't covering the intent either, maybe rename to `getOrCreateOAuthClientService`? Depending on the feedback I'll update the javadoc also.

Reply:
I still try to understand the reason for this refactoring. Before the factory did create the handle which you used to access the corresponding client. now you provide the handle yourself but may overwrite an existing client with new parameters. What issue are you solving with this?

Reply:
It fixes the following 2 problems:
- if the method generates the handle string the binding must store this, which would be done in a property or parameter. That is not very convenient, so if the binding provides a handle itself, like the thing uid, it doesn't require some quirky way to remember the handle.
- There was no way to update the config in case a value changes. For example if clientid would change there is no way to check this. Which basically would mean that every time the handle must be recreated because there is no way to know for sure the handle contains the correct values. Not doing this would result in unexpected behavior.

So in a binding this method will always be used to get the handle and makes sure it will have up-to-date information. I understand my implementation does a lot of things in one method, but the other solution I could think of would be to add several methods and then probably all users would have to implement a similar pattern, while usage would probably not much different I think.But this would be a good discussion point.

Reply:
Thanks @Hilbrand. 
Please @sobimibos or @garytse pick this one up.